how-to-guides/12-train-multi-output-model.ipynb

```diff
@@ -0,0 +1,410 @@
+        "\n",
+        "    def __len__(self):\n",
+        "        return self.end - self.start + 1\n",
+        "\n",
+        "    def __getitem__(self, idx):\n",
+        "        if torch.is_tensor(idx):\n",
+        "            idx = idx.tolist()\n",
+        "\n",
+        "        item = self.start + idx\n",
+        "        y1 = power2(item)\n",
+        "        y2 = mul2(item)\n",
+        "        sample = [torch.Tensor([item]), torch.Tensor([y1]), torch.Tensor([y2])]\n",
+        "        return sample\n",
+        "\n",
+        "train_dataset = RangeDataset(1, 80)\n",
+        "val_dataset = RangeDataset(90, 100)\n",
+        "\n",
+        "train_loader = DataLoader(train_dataset, batch_size=4, shuffle=True)\n",
+        "val_loader = DataLoader(val_dataset, batch_size=4, shuffle=False)"
+      ],
+      "id": "RQARkoCcNO1o",
+      "execution_count": 34,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Gynzy2VWOpRL"
+      },
+      "source": [
+        "## Model\n",
+        "\n",
+        "Let's create a dummy `Net()`. We have two linear layers for predicting the two labels which we can simply return from `forward()`."
+      ],
+      "id": "Gynzy2VWOpRL"
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "iK_9cOP6a1TI"
+      },
+      "source": [
+        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
+        "\n",
+        "class Net(nn.Module):\n",
+        "\n",
+        "    def __init__(self):\n",
+        "        super(Net, self).__init__()\n",
+        "        \n",
+        "        self.model1 = nn.Linear(1, 1)\n",
+        "        self.model2 = nn.Linear(1, 1)\n",
+        "\n",
+        "    def forward(self, x):\n",
+        "        return self.model1(x), self.model2(x)\n",
+        "\n",
+        "model = Net().to(device)\n",
+        "optimizer = torch.optim.Adam(model.parameters(), lr=0.005)\n",
+        "criterion = nn.MSELoss()"
+      ],
+      "id": "iK_9cOP6a1TI",
+      "execution_count": 35,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ODT-jag3O8B1"
+      },
+      "source": [
+        "## Trainer\n",
+        "\n",
+        "Now we will create a custom `trainer` which unpacks a batch, makes a prediction, calculates the two losses and returns their sum."
+      ],
+      "id": "ODT-jag3O8B1"
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ItoswUK-23St"
+      },
+      "source": [
+        "def train_step(engine, batch):\n",
+        "    model.train()\n",
+        "    optimizer.zero_grad()\n",
+        "    x, y1, y2 = batch[0].to(device), batch[1].to(device), batch[2].to(device)\n",
+        "    y_pred1, y_pred2 = model(x)\n",
+        "    loss1 = criterion(y_pred1, y1)\n",
+        "    loss2 = criterion(y_pred2, y2)\n",
+        "\n",
+        "    loss1.backward()\n",
```

Comment:
We talked about the tracking of each loss' components to figure out that `x^2` can't be approximate by a `Linear` but `2*x` can. 
```python
loss = loss1 + loss2
loss.backward()

return {"loss": loss.item(), "loss1": loss1.item(), "loss2": loss2.item()}
```